### PR TITLE
add edit feature page

### PR DESCRIPTION
**Pull request by ylwu-amzn**
This is part of this big PR https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/40, the big PR was closed and will publish some smaller PRs to make it easy to review.

*Issue #, if available:*

*Description of changes:*

* Add edit feature page
* Support preview sample anomalies

screencapture-localhost-5601-app-opendistro-anomaly-detection-2020-04-17-10_13_51

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/pages/EditFeatures/components/CustomAggregation/CustomAggregation.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    return 'Invalid JSON';
+  }
+};
+
+export const CustomAggregation = (props: CustomAggregationProps) => {
+  return (
+    <Field
+      id={`featureList.${props.index}.aggregationQuery`}
+      name={`featureList.${props.index}.aggregationQuery`}
+      validate={validateQuery}
```
Comment (yizheliu-amazon) [inline public/pages/EditFeatures/components/CustomAggregation/CustomAggregation.tsx:42]: what will happen if `Invalid JSON` is returned by `validateQuery`?

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+import { formikToSimpleAggregation } from '../../containers/utils/formikToFeatures';
+import { AggregationSelector } from '../AggregationSelector';
+import { CustomAggregation } from '../CustomAggregation';
+
+interface FeatureAccordionProps {
+  onDelete(): void;
+  index: number;
+  feature: any;
+  handleChange(event: React.ChangeEvent<HTMLSelectElement>): void;
+  // ref: any;
```
Comment (yizheliu-amazon) [inline public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:43]: remove this

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    </Fragment>
+  );
+
+  const customAggDescription = (feature: any) => (
+    <Fragment>
+      <span className="content-panel-subTitle" style={{ paddingRight: '20px' }}>
+        Custom expression
+      </span>
+      <span className="content-panel-subTitle">
+        State: {feature.featureEnabled ? 'Enabled' : 'Disabled'}{' '}
```
Comment (yizheliu-amazon) [inline public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:70]: not used `{' '}`?

File: public/pages/EditFeatures/components/AggregationSelector/__tests__/AggregationSelectior.test.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+/*
```
Comment (ohltyler) [inline public/pages/EditFeatures/components/AggregationSelector/__tests__/AggregationSelectior.test.tsx:1]: typo: rename file to AggregationSelector.test.tsx

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+              </h5>
+            </EuiTitle>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        {showSubtitle ? showFeatureDescription(feature) : null}
+      </div>
+    );
+  };
+
+  const extraAction = (onClick: any) => (
```
Comment (yizheliu-amazon) [inline public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:98]: rename to `deleteAction`?

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        {({ field, form }: FieldProps) => (
+          <EuiFormRow
+            label="Feature name"
+            helpText="Enter a descriptive name. The name must be unique within this detector. Feature name must contain 1-64 characters. Valid characters are a-z, A-Z, 0-9, -(hyphen) and _(underscore)."
+            isInvalid={isInvalid(field.name, form)}
+            error={getError(field.name, form)}
+          >
+            <EuiFieldText
+              name={`featureList.${props.index}.featureName`}
+              // id={`featureList.${props.index}.featureName`}
```
Comment (yizheliu-amazon) [inline public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:134]: remove

File: public/pages/EditFeatures/utils/helpers.ts [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+  if (
+    //@ts-ignore
+    !!get(errors, 'featureList', []).filter(featureError => featureError).length
+  ) {
+    const featureList = get(errors, 'featureList', []);
+    for (let i = featureList.length - 1; i >= 0; i--) {
+      if (featureList[i]) {
+        forOwn(featureList[i], function(value, key) {
+          debugger;
+          console.log(value, key);
```
Comment (yizheliu-amazon) [inline public/pages/EditFeatures/utils/helpers.ts:139]: may remove this.

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+
+  const extraAction = (onClick: any) => (
+    <EuiButton size="s" color="danger" onClick={onClick}>
+      Delete
+    </EuiButton>
+  );
+
+  return (
+    <EuiAccordion
+      // ref={props.ref}
```
Comment (ohltyler) [inline public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:106]: can remove this

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx (hunk 6 above)
Comment (ohltyler) [inline public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:134]: can remove this

File: public/pages/EditFeatures/components/CustomAggregation/CustomAggregation.tsx (hunk 1 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/components/CustomAggregation/CustomAggregation.tsx:42]: We will show error below the code editor like "Invalid JSON" and will block saving and previewing .

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx (hunk 2 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:43]: sure, will remove

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx (hunk 3 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:70]: Will remove it

File: public/pages/EditFeatures/components/AggregationSelector/__tests__/AggregationSelectior.test.tsx (hunk 4 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/components/AggregationSelector/__tests__/AggregationSelectior.test.tsx:1]: sure, will change

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx (hunk 5 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:98]: sure, will change

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx (hunk 6 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:134]: sure

File: public/pages/EditFeatures/utils/helpers.ts (hunk 7 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/utils/helpers.ts:139]: sure

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx (hunk 8 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:106]: sure

File: public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx (hunk 6 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/components/FeatureAccordion/FeatureAccordion.tsx:134]: sure

**Pull request merged**